### ISSUE TITLE
ResultSet.getString() should work for non-String columns

### ISSUE THREAD

**Issue by DoodleBobBuffPants** [googleapis/java-spanner-jdbc#195]
**Is your feature request related to a problem? Please describe.**

When calling `ResultSet.getString()`, it fails in Spanner for non-String columns, whereas every other database JDBC driver will convert the column to String for you.

Our callstack is:

```
java.lang.IllegalStateException: Column 0 is not of correct type: expected STRING but was INT64

	at com.google.common.base.Preconditions.checkState(Preconditions.java:854)
	at com.google.cloud.spanner.AbstractStructReader.checkNonNullOfType(AbstractStructReader.java:342)
	at com.google.cloud.spanner.AbstractStructReader.getString(AbstractStructReader.java:132)
	at com.google.cloud.spanner.ForwardingStructReader.getString(ForwardingStructReader.java:112)
	at com.google.cloud.spanner.jdbc.DirectExecuteResultSet.getString(DirectExecuteResultSet.java:176)
	at com.google.cloud.spanner.jdbc.JdbcResultSet.getString(JdbcResultSet.java:126)
```

**Describe the solution you'd like**

The callstack shows that the driver knows what the actual column type is. Could the driver get it as the correct type and return the value converted to String itself?

**Describe alternatives you've considered**

The alternative we considered is switching on every type based on the metadata but this isn't required on any other database driver and bloats the code when the driver can do it internally.

**Comment by olavloite** [comment]
@DoodleBobBuffPants Which version are you currently using? As this issue should have been fixed in version 1.16.0 [by this PR](https://github.com/googleapis/java-spanner-jdbc/pull/143).

**Comment by DoodleBobBuffPants** [comment]
Using 1.16.0 worked! Thanks!
